geopandas/base.py

```diff
@@ -2574,6 +2574,117 @@ def hausdorff_distance(self, other, align=True, densify=None):
         """
         return _binary_op("hausdorff_distance", self, other, align, densify=densify)
 
+    def frechet_distance(self, other, align=True, densify=None):
+        """Returns a ``Series`` containing the Frechet distance to aligned `other`.
+
+        The Frechet distance is the largest distance consisting of any vertex in `self`
+        with the nearest vertex in `other`.
```

Comment:
This is not true. Frechet distance is not defined using vertices. This implementation is considering only vertices. That is different. I'd use the text from shapely docstring directly.